File: uikit/src/main/java/io/agora/chat/uikit/conversation/adapter/EaseConversationListAdapter.java
package io.agora.chat.uikit.conversation.adapter;


import android.view.LayoutInflater;
import android.view.ViewGroup;

import io.agora.chat.uikit.adapter.EaseBaseRecyclerViewAdapter;
import io.agora.chat.uikit.conversation.model.EaseConversationInfo;
import io.agora.chat.uikit.conversation.model.EaseConversationSetStyle;
import io.agora.chat.uikit.conversation.viewholder.EaseConversationType;
import io.agora.chat.uikit.conversation.viewholder.EaseConversationViewHolderFactory;

public class EaseConversationListAdapter extends EaseBaseRecyclerViewAdapter<EaseConversationInfo> {
    private EaseConversationSetStyle style;

    public EaseConversationListAdapter() {}

    public EaseConversationListAdapter(EaseConversationSetStyle style) {
        this.style = style;
    }

    @Override
    public int getItemNotEmptyViewType(int position) {
        return EaseConversationViewHolderFactory.getViewType(mData.get(position));
    }

    @Override
    public ViewHolder<EaseConversationInfo> getViewHolder(ViewGroup parent, int viewType) {
        return EaseConversationViewHolderFactory.createViewHolder(LayoutInflater.from(parent.getContext())
                , parent, EaseConversationType.from(viewType), style);
    }

    public void setConversationSetStyle(EaseConversationSetStyle style) {
        this.style = style;
        notifyDataSetChanged();
    }

}

File: ease-im-kit/src/main/kotlin/io/agora/uikit/feature/conversation/adapter/EaseConversationListAdapter.kt
package io.agora.uikit.feature.conversation.adapter

import android.view.LayoutInflater
import android.view.ViewGroup
import io.agora.uikit.base.EaseBaseRecyclerViewAdapter
import io.agora.uikit.feature.conversation.config.EaseConvItemConfig
import io.agora.uikit.feature.conversation.viewholders.EaseConvViewType
import io.agora.uikit.feature.conversation.viewholders.EaseConversationViewHolderFactory
import io.agora.uikit.model.EaseConversation

/**
 * Adapter for conversation list.
 * @param config Conversation item config.
 */
open class EaseConversationListAdapter(
    private var config: EaseConvItemConfig = EaseConvItemConfig()
): EaseBaseRecyclerViewAdapter<EaseConversation>() {

    override fun getItemNotEmptyViewType(position: Int): Int {
        return EaseConversationViewHolderFactory.getViewType(getItem(position))
    }
    override fun getViewHolder(parent: ViewGroup, viewType: Int) =
        EaseConversationViewHolderFactory.createViewHolder(LayoutInflater.from(parent.context), parent,
            EaseConvViewType.fromValue(viewType), config)

    /**
     * Set conversation item config.
     */
    fun setConversationItemConfig(config: EaseConvItemConfig) {
        this.config = config
        notifyDataSetChanged()
    }
}